### PR TITLE
add compatibility matrix to the docs

### PR DESCRIPTION
**Pull request by m-vdb**
**Proposed changes**:
- Add compatibility matrix (from decision on [this Notion page](https://www.notion.so/rasa/Rasa-Pro-Services-Versioning-01133b6faba143248541a1dfa75e20da))

### CODE REVIEW

File: docs/docs/installation/rasa-pro/installation.mdx [hunk 1]
```diff
@@ -19,6 +19,9 @@ To learn more about deployments, please visit the following sections:
 - [Deploying Rasa Plus in production](../../deploy/deploy-rasa-plus)
 - [Deploying Rasa Pro Services in production](../../deploy/deploy-rasa-pro-services)
 
+To learn more about which version of Rasa Pro Services is compatible with Rasa Plus, please head over to the
+[compatibility matrix](../../compatibility-matrix.mdx).
```
Comment (m-vdb) [inline docs/docs/installation/rasa-pro/installation.mdx:23]: not sure if it's the best place for that plug. LMK!

File: docs/docs/compatibility-matrix.mdx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+id: compatibility-matrix
```
Comment (chandrikas) [inline docs/docs/compatibility-matrix.mdx:2]: I think we should also add to this matrix https://rasa.com/docs/rasa-enterprise/changelog/compatibility-matrix/

File: docs/docs/installation/rasa-pro/installation.mdx [hunk 3]
```diff
@@ -19,6 +19,9 @@ To learn more about deployments, please visit the following sections:
 - [Deploying Rasa Plus in production](../../deploy/deploy-rasa-plus)
 - [Deploying Rasa Pro Services in production](../../deploy/deploy-rasa-pro-services)
 
+To learn more about which version of Rasa Pro Services is compatible with Rasa Plus, please head over to the
```
Comment (chandrikas) [inline docs/docs/installation/rasa-pro/installation.mdx:22]:
```suggestion
To learn more about compatibility between different versions of  Rasa Plus and Rasa Pro Services, please see
```

File: docs/docs/installation/rasa-pro/installation.mdx (hunk 1 above)
Comment (chandrikas) [reply docs/docs/installation/rasa-pro/installation.mdx:23]: I think the place is right

File: docs/docs/compatibility-matrix.mdx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+
+When choosing a version of Rasa Pro Services to [deploy](./deploy/deploy-rasa-pro-services.mdx), you can refer to the
+table below to see which one is compatible with your installation of Rasa Plus.
+
+Rasa Pro Services version is independent of Rasa Plus version, except that they share the same major version number.
+
+
+| Rasa Pro Services | Rasa Plus |
```
Comment (chandrikas) [inline docs/docs/compatibility-matrix.mdx:16]: This doesn't render correctly in the preview for some reason 🤔

Comment (m-vdb) [reply docs/docs/compatibility-matrix.mdx:16]: yeah I think the theme tables are broken everywhere... will see if we can address this before the release (I am not sure we can)

**Pull request merged**